### ISSUE TITLE
watchdog: automated registration

### ISSUE THREAD

**Issue by carueda** [mmisw/orr-ont#41]
In old system: https://github.com/mmisw/mmiorr/issues/251

I think the initial implementation approach should/could be in separate tools, like [cf2rdf](https://github.com/mmisw/cf2rdf) (specifically for CF), and our most recent one, [orr-reg](https://github.com/mmisw/orr-reg), which currently is for SWEET but could be made more generic. Perhaps after some initial working prototypes, we can decide on how to incorporate or integrate some of the functionality with the ORR.

**Comment by graybeal** [comment]
I still think this is a significant limitation of the system (or to put it more positively, would be a huge plus for the system), and is not terribly difficult to implement in a fully generic way. (If attribute autoupdate is true, download the ontology resource specified by the Original Source (or similar attribute), and if different from existing download (might have to keep and check against a checksum), re-import.  

I'm not arguing for implementing it now, just emphasizing how valuable I think it will be.

**Comment by carueda** [comment]
Ok, so I just marked it high priority (not sure about deadline, though)

**Comment by graybeal** [comment]
Yes, no deadline for this, there is no immediate user requirement.

**Comment by carueda** [comment]
Closely related: #39 

An option to specify a URL (instead of local file) will be incorporated in the first page of the current "Upload" sequence. 

When selecting a URL, the user could indicate an number of possible "watchdog" options:

- Automated registration of new versions:
  - `(√ )` Do not do perform
  - `( )` Perform automated processing:
    - '( )' Check for updates every (.. day, week, ...)
    - `( )` Use webhook  _(an ORR webhook URL will be presented for the user to capture in external service associated with the remote ontology)_

- When automatically registering a new version:
  - **Status**:  
    - `(√ )` Assign current status   
    - `( )`  Assign the following: `( ) draft    ( )  testing ....`
  - **Visibility**:  
    - `(√ )` Assign current  visibility   
    - `(  )` Assign the following: `( ) owner    ( ) public`

**Comment by graybeal** [comment]
I don't see a significant value in the web hook option (well, very nice for CF and maybe SWEET or others, I see how that would be nice -- but not essential if it's hard), and definitely not for the ability to set status on auto-uploads.


It is a given that the resource at the named URL is under the control of the ontology provider, so whatever status they give the ontology originally, the rest of them will get the same status.

________________________________

**Comment by pbuttigieg** [comment]
Greetings all! ENV:earth_africa:  developer here. 

Auto-synchronising with ontology releases is very essential. As you probably know, tools like [OntoBee ](http://www.ontobee.org/) and EBI's [OLS](https://www.ebi.ac.uk/ols/index) have good auto-updating approaches which make sure there's no version (and therefore semantic) drift. 

We (and other OBO ontologies) are using GitHub tagged releases when we update: https://github.com/EnvironmentOntology/envo/releases

**Comment by carueda** [comment]
Thanks Pier.

As discussed yesterday (offline), and given that the automated havesting is still a TODO, I'm just going to remove this entry from the COR. Then, if you'd like, you can go ahead and exercise uploading ENVO and let us know how it goes.   Ah, a related issue entry is https://github.com/mmisw/orr-portal/issues/62

**Comment by carueda** [comment]
Unregistration complete.
Please go ahed with the new registration at your convenience.  New organization created, http://cor.esipfed.org/ont/#/org/obo , with plbuttigieg as member.  Thanks!

**Comment by carueda** [comment]
(oops, seems like I clicked the wrong button before -- this is still open)

**Comment by lewismc** [comment]
Hi @carueda I think the approach you've highlighted in https://github.com/mmisw/orr-ont/issues/41#issuecomment-280550297 makes perfect sense.
